### PR TITLE
Add Matroska support to file type resolver

### PR DESCRIPTION
**Pull request by sandsmark**
Updated the constructors to match the other file types, and added .mka and .mkv as recognized file types.

### CODE REVIEW

**Comment by sandsmark** [comment]
Tested it with the tagreader and inspect examples.

File: taglib/fileref.cpp [hunk 1]
```diff
@@ -52,6 +52,7 @@
 #include "itfile.h"
 #include "xmfile.h"
 #include "dsffile.h"
+#include "ebml/matroska/ebmlmatroskafile.h"
```
Comment (FestusHagen) [inline taglib/fileref.cpp]: TagLib style, Is without path.

File: taglib/fileref.cpp [hunk 2]
```diff
@@ -181,6 +182,9 @@ namespace
       return new XM::File(arg, readAudioProperties, audioPropertiesStyle);
     if (ext == "DSF")
       return new DSF::File(arg, readAudioProperties, audioPropertiesStyle);
+    if (ext == "MKA" || ext == "MKV") {
```
Comment (FestusHagen) [inline taglib/fileref.cpp]: TagLib style is without spaces between function and opening parenthesis.

If you would, Please also fix the DSF one that was previously missed right above as well ... Thank you!

File: taglib/ebml/matroska/ebmlmatroskafile.cpp [hunk 3]
```diff
@@ -134,7 +134,7 @@ EBML::Matroska::File::~File()
   }
 }
 
-EBML::Matroska::File::File(FileName file) : EBML::File(file), d(0)
+EBML::Matroska::File::File(FileName file, bool, AudioProperties::ReadStyle) : EBML::File(file), d(0)
```
Comment (FestusHagen) [inline taglib/ebml/matroska/ebmlmatroskafile.cpp]: Should be `bool readProperties` ... ?

File: taglib/ebml/matroska/ebmlmatroskafile.cpp [hunk 4]
```diff
@@ -145,7 +145,7 @@ EBML::Matroska::File::File(FileName file) : EBML::File(file), d(0)
   }
 }
 
-EBML::Matroska::File::File(IOStream *stream) : EBML::File(stream), d(0)
+EBML::Matroska::File::File(IOStream *stream, bool, AudioProperties::ReadStyle) : EBML::File(stream), d(0)
```
Comment (FestusHagen) [inline taglib/ebml/matroska/ebmlmatroskafile.cpp]: Again `bool readProperties` ... ?

File: taglib/ebml/matroska/ebmlmatroskafile.h [hunk 5]
```diff
@@ -47,14 +48,28 @@ namespace TagLib {
...
+         * Constructs a Matroska file from \a stream.  If \a readProperties is true the
+         * file's audio properties will also be read using \a propertiesStyle.  If
+         * false, \a propertiesStyle is ignored.
+         *
+         * \note In the current implementation, both \a readProperties and
+         * \a propertiesStyle are ignored.  The audio properties are always
+         * read.
          */
-        File(IOStream *stream);
+        File(IOStream *stream, bool readproperties = true,
```
Comment (FestusHagen) [inline taglib/ebml/matroska/ebmlmatroskafile.h]: `bool read**P**roperties` ...

**Comment by AkshayBudhiraja** [comment]
Hello, any info on when this pull request is planned on being merged? I would like to use this to write metadata tags to MKV video files.

**Comment by sandsmark** [comment]
rebased and fixed the comments, but the dsf stuff seems gone?

**Comment by willkara** [comment]
Any updates?

**Comment by sandsmark** [comment]
rebased and fixed the include

**Comment by sbooth** [comment]
I manually merged this PR.

**Pull request closed without merging**